hazelcast/src/main/java/com/hazelcast/internal/partition/impl/InternalPartitionServiceImpl.java

```diff
@@ -691,7 +689,62 @@ public InternalPartition getPartition(int partitionId, boolean triggerOwnerAssig
 
     @Override
     public boolean prepareToSafeShutdown(long timeout, TimeUnit unit) {
-        return partitionReplicaStateChecker.prepareToSafeShutdown(timeout, unit);
+        if (!node.joined()) {
+            return true;
+        }
+
+        if (node.isLiteMember()) {
+            return true;
```

Comment:
answering myself: handling tasks running by exec service is out of scope. this pr handles migration system.